src/util.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Makes first character in str upper case
+ *
+ * @param {string} str
+ * @returns {string}
+ */
+export function capitalize(str) {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+/**
+ * Get tags from all lessons in lessonContext
+ * @param lessonContext
+ * @returns {Object} tags
+ */
+export function getTags(lessonContext) {
+  const paths = lessonContext.keys();
+
+  return paths.reduce((res, path) => {
+    const lessonFrontMatter = lessonContext(path).frontmatter;
+    const tags = cleanseTags(lessonFrontMatter.tags);
+    res = mergeObjects(tags, res);
+    return res;
+  }, {});
+}
+
+/**
+ * Get list of all courses in lessonContext
+ * @param lessonContext
+ * @param iconContext
+ * @returns {Array}
+ */
+export function getCourses(lessonContext, iconContext) {
+  const paths = lessonContext.keys();
+
+  return paths.reduce((res, path) => {
+    const courseName = path.slice(2, path.indexOf('/', 2));
+    const lessonFrontMatter = lessonContext(path).frontmatter;
+    const tags = cleanseTags(lessonFrontMatter.tags);
+    const lesson = {name: lessonFrontMatter.title, tags: tags};
+    const index = res.findIndex(course => course.name === courseName);
+
+    // If course already exists, push the new lesson. Else make a new course
+    if(index >= 0) {
+      res[index].lessons.push(lesson);
+    }else{
+      res.push({
+        lessons: [lesson],
+        name: courseName,
+        iconPath: iconContext('./' + courseName + '/logo-black.png'),
+        path: path
+      });
+    }
+
+    return res;
+  }, []);
+}
+
+/**
+ * Will combine lists in same properties from objA and objB. The lists will be combined as a set of distinct elements
+ * @param {Object} objA
+ * @param {Object} objB
+ * @returns {Object} merged object
+ */
+export function mergeObjects(objA, objB) {
+  const allProps = [...new Set(Object.keys(objA).concat(Object.keys(objB)))];
+  return allProps.reduce((res, propName) => {
+    const listItemsA = objA.hasOwnProperty(propName) ? objA[propName] : [];
+    const listItemsB = objB.hasOwnProperty(propName) ? objB[propName] : [];
+    res[propName] = [...new Set(listItemsA.concat(listItemsB))];
+    return res;
+  }, {});
+}
+
+/**
+ * Add tag to filter
+ * @param {Object} tag
+ * @param {Object} filter
+ * @returns {Object} filter with tag merged into it
+ */
+export function addTagToFilter(tag, filter) {
+  return mergeObjects(filter, tag);
```

Comment:
Tanken var å gjøre koden mer lesbar. 

`const filter = checked ? addTagToFilter(tag, this.state.filter) : removeTagFromFilter(tag, this.state.filter);`

Jeg har prøvd å holde FrontPage så liten og lesbar som mulig, slik at man raskt får oversikt over hva som skjer der. Alle implementasjonsdetaljer er flyttet til util.js.
Skal selvfølgelig endre til mergeObjects hvis det gjør koden enklere å forstå